core/src/main/scala/org/http4s/Uri.scala

```diff
@@ -280,6 +279,110 @@ object Uri {
     }
   }
 
+  final class Path(
+      val segments: Chain[String],
+      val absolute: Boolean = false,
+      val endsWithSlash: Boolean = false)
+      extends Renderable {
+
+    def isEmpty: Boolean = segments.isEmpty
+    def nonEmpty: Boolean = segments.nonEmpty
+
+    override def equals(obj: Any): Boolean = obj match {
+      case s: String => doEquals(Uri.Path.fromString(s))
+      case p: Path => doEquals(p)
+      case _ => false
+    }
+
+    private def doEquals(path: Path): Boolean =
+      this.segments == path.segments && path.absolute == this.absolute && path.endsWithSlash == this.endsWithSlash
+
+    override def hashCode(): Int = 31 * segments.hashCode()
```

Comment:
Why not acccount for absolute and endsWithSlash here, too?  You don't _have_ to, but you could.

Reply:
forgot that, so will add it